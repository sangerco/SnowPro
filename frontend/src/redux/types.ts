export interface UserData {
    username: string;
    firstName: string;
    lastName: string;
    email: string;
    avatar: string;
    bio: string;
    videos: string[];
    photos: string[];
    favMountains: string[];
  }


export const FETCH_USER_DATA_REQUEST = 'FETCH_USER_DATA_REQUEST';
export const FETCH_USER_DATA_SUCCESS = 'FETCH_USER_DATA_SUCCESS';
export const FETCH_USER_DATA_FAILURE = 'FETCH_USER_DATA_FAILURE';


export interface FetchUserDataRequestAction {
    type: typeof FETCH_USER_DATA_REQUEST
}

export interface FetchUserDataSuccessAction {
    type: typeof FETCH_USER_DATA_SUCCESS;
    payload: UserData;
}

export interface FetchUserDataFailureAction {
    type: typeof FETCH_USER_DATA_FAILURE;
    payload: string;
}

export type UserActionTypes = 
    FetchUserDataRequestAction
    | FetchUserDataSuccessAction
    | FetchUserDataFailureAction;