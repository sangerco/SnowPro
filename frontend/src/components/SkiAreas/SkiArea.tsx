import React, { useState, useEffect } from "react";
import axios from "axios";
import { URL } from "../../utils/config";
import { Link } from "react-router-dom";
import { Container, Header, Icon } from "semantic-ui-react";
import { SkiAreaProps, SkiAreaData } from "../../interfaces/skiAreaInterfaces";

const SkiArea: React.FC<SkiAreaProps> = ({
  slug,
  name,
  country,
  region,
  lat,
  long,
  url,
}) => {
  const [skiAreaDataObject, setSkiAreaDataObject] = useState({ href: "" });

  useEffect(() => {
    const fetchSkiAreaData = async () => {
      try {
        const response: SkiAreaData = await axios.get(
          `${URL}/ski-areas/${slug}`
        );
        setSkiAreaDataObject(response.data);
      } catch (e: any) {
        const errorMessage = e.message;
        console.error(e);
        throw new Error(`Error Loading Ski Area! ${errorMessage}`);
      }
    };

    fetchSkiAreaData();
  }, [slug]);

  const href = skiAreaDataObject.href;

  console.log(slug);

  return (
    <Container text>
      <Header size="huge">
        <Link to={`/ski-areas/${slug}`}>{name}</Link>
      </Header>
      <Container textAlign="right">
        <p style={{ color: "royalblue" }}>{country}</p>
        <p style={{ color: "royalblue" }}>{region}</p>
        {/* <a style={{ color: "royalblue" }} href={url}>{ url }</a> */}
      </Container>
      <Container textAlign="right">
        <p>
          <small>
            <span style={{ color: "royalblue" }}>
              <Icon name="map outline" />
            </span>{" "}
            {lat}, {long}
          </small>
        </p>
      </Container>
    </Container>
  );
};

export default SkiArea;
