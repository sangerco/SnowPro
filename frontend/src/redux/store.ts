import { Action, compose } from "redux";
import { ThunkAction, configureStore } from "@reduxjs/toolkit";
import { logger } from "redux-logger";
import rootReducer from "./reducers/rootReducer";

const composeEnhancers =
  (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(logger, composeEnhancers),
});
