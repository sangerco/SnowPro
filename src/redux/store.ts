import { configureStore, combineReducers } from "@reduxjs/toolkit";
import userReducer from "./slices/userSlice";
import authReducer from "./slices/authSlice";
import messageReducer from "./slices/messageSlice";
import messageReplyReducer from "./slices/messageReplySlice";
import reviewReducer from "./slices/reviewSlice";
import reviewReplyReducer from "./slices/reviewReplySlice";
import tagReducer from "./slices/tagSlice";
import storage from "redux-persist/lib/storage";
import { persistReducer, persistStore } from "redux-persist";

const persistConfig = {
  key: "root",
  storage,
  blacklist: ["messages", "messageReplies", "reviews", "reviewReplies"],
};

const rootReducer = combineReducers({
  users: userReducer,
  auth: authReducer,
  messages: messageReducer,
  messageReplies: messageReplyReducer,
  reviews: reviewReducer,
  reviewReplies: reviewReplyReducer,
  tags: tagReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const persistor = persistStore(store);
