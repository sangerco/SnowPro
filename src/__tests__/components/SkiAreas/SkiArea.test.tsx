/* eslint-disable testing-library/no-node-access */
import React from "react";
import SkiArea from "../../../components/SkiAreas/SkiArea";
import { render, fireEvent, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../../../redux/store";
import { BrowserRouter } from "react-router-dom";

describe("test Ski Area", () => {
  it("should render without crashing", () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <SkiArea
            slug={"ski-area-1"}
            name={"Test Ski Area"}
            country={"US"}
            region={"CO"}
            lat={0}
            long={0}
            url={"https://testskiarea.com"}
          />
        </BrowserRouter>
      </Provider>
    );
  });
  it("should match snapshot", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <SkiArea
            slug={"ski-area-1"}
            name={"Test Ski Area"}
            country={"US"}
            region={"CO"}
            lat={0}
            long={0}
            url={"https://testskiarea.com"}
          />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("should show ski area data", () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <SkiArea
            slug={"ski-area-1"}
            name={"Test Ski Area"}
            country={"US"}
            region={"CO"}
            lat={0}
            long={0}
            url={"https://testskiarea.com"}
          />
        </BrowserRouter>
      </Provider>
    );

    const header = screen.getByTestId("ski-area-header");
    expect(header).toBeInTheDocument();

    const name = screen.getByTestId("ski-area-name");
    expect(name.textContent).toEqual("Test Ski Area");

    const country = screen.getByTestId("ski-area-country");
    expect(country.textContent).toEqual("US");

    const region = screen.getByTestId("ski-area-region");
    expect(region.textContent).toEqual("CO");
  });
});
