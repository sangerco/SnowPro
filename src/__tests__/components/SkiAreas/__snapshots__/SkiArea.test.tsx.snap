// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Ski Area should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="ui raised segment"
  >
    <div
      class="ui huge header"
      data-testid="ski-area-header"
    >
      <a
        data-testid="ski-area-name"
        href="/ski-areas/ski-area-1"
      >
        Test Ski Area
      </a>
    </div>
    <div
      class="ui right aligned container"
    >
      <p
        data-testid="ski-area-country"
        style="color: royalblue;"
      >
        US
      </p>
      <p
        data-testid="ski-area-region"
        style="color: royalblue;"
      >
        CO
      </p>
    </div>
    <div
      class="ui right aligned container"
    >
      <p>
        <small>
          <span
            style="color: royalblue;"
          >
            <i
              aria-hidden="true"
              class="map outline icon"
            />
          </span>
           0, 0
        </small>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
